Assets/MixedRealityToolkit/_Core/Interfaces/Diagnostics/IMixedRealityDiagnosticsManager.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Diagnostics;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.Events;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Interfaces.Diagnostics
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public interface IMixedRealityDiagnosticsManager : IMixedRealityEventSystem, IMixedRealityEventSource
+    {
+        /// <summary>
+        /// Enable / disable the diagnostic display
+        /// </summary>
+        bool Visible { get; }
+
+        /// <summary>
+        /// Enable / disable cpu profiling when the diagnostic panel is visible. 
+        /// </summary>
+        bool ShowCpu { get; }
+
+        /// <summary>
+        /// Enable / disable fps profiling when the diagnostic panel is visible. 
+        /// </summary>
+        bool ShowFps { get; } 
+
+        /// <summary>
+        /// Enable / disable memory profiling when the diagnostic panel is visible. 
+        /// </summary>
+        bool ShowMemory { get; }
+
+        /// <summary>
+        /// Gets the <see cref="GameObject"/> that represents the diagnostic visualization
+        /// </summary>
+        GameObject DiagnosticVisualization { get; }
```

Comment:
Do we want to agree on a naming convention here?

There's been talk of using proxy vs visualization

Reply:
Sorry, I haven't been following this PR/issue. What is DiagnosticVisualization for? I'm not sure how this relates to ControllerProxies (or another kind of scene proxy?).

Reply:
It's a naming pattern we've already established using visualization vs proxy as the game object reference.

I do like proxy a bit more, as it's just a reference to the game object, not the things being rendered (the renderer component comes to mind here)

Reply:
I'm fine with renaming to Proxy if that's the pattern we have.

Reply:
I'd only name it a Proxy if it's a game object proxying something in the device layer. If it's just a game object for visualizing diagnostic data, I think Visualization or Visuals or Visualizer are all fine names.

Reply:
Just trying to figure out what pattern we want to stick with. 🤔

Reply:
They're both game object references.

I just wanna make sure we're on the same page about how we want to talk about and reference these things.

Reply:
>They're both game object references.

I don't think that was the important part of the original suggestion. It was pretty specific to the controller scenario (and probably a small number of others), but I don't see the relationship here.

Reply:
Naming is pretty important to the idea of a concept or thing. 

I don't really see a difference here.
I see them as pretty much the same concepts.

We have a game object reference that doesn't actually do any "visualization" but it's a stand in (proxy) for the concept as a whole.

If we wanted to be specific that it's actually "visualizing" something the field type should reflect that by being a renderer or something that does the rendering in the scene.

Reply:
The naming isn't about GameObject vs renderer vs transform vs anything else.

It's about physical device/object vs text.

As Eric said above:

>I'd only name it a Proxy if it's a game object proxying something that lives in the device layer. If it's just a game object for visualizing diagnostic data, I think Visualization or Visuals or Visualizer are all fine names.

Reply:
That's the thing tho. It's not text. It's a game object.

Reply:
Right, and, again, it's not about the fact that they're both GameObjects. It's about the things those GameObjects are representing, which are very different.

Reply:
But they're similar in the fact that they reference a thing in the scene, and that reference doesn't actually do any "visualization" or rendering of said thing.

Reply:
Are you proposing that all GameObject references in code should be appended with `Proxy`?

Reply:
No, just the ones that have "visualization" or similar as the name.

Reply:
I created a new PR from a branch I can edit to handle merge conflicts. I am going to abandon this one, so please continue any discussions in #2846